src/Symfony/Component/Security/Core/Tests/Authorization/UserAuthorizationCheckerTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Tests\Authorization;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Security\Core\Authentication\Token\UserAuthorizationCheckerToken;
+use Symfony\Component\Security\Core\Authorization\AccessDecisionManagerInterface;
+use Symfony\Component\Security\Core\Authorization\UserAuthorizationChecker;
+use Symfony\Component\Security\Core\User\InMemoryUser;
+
+class UserAuthorizationCheckerTest extends TestCase
+{
+    private AccessDecisionManagerInterface&MockObject $accessDecisionManager;
+    private UserAuthorizationChecker $authorizationChecker;
+
+    protected function setUp(): void
+    {
+        $this->accessDecisionManager = $this->createMock(AccessDecisionManagerInterface::class);
+
+        $this->authorizationChecker = new UserAuthorizationChecker($this->accessDecisionManager);
+    }
+
+    /**
+     * @dataProvider isGrantedProvider
+     */
+    public function testIsGranted(bool $decide)
+    {
+        $token = new UserAuthorizationCheckerToken(new InMemoryUser('username', 'password', ['ROLE_USER']));
+
+        $this->accessDecisionManager
+            ->expects($this->once())
+            ->method('decide')
+            ->with($this->isInstanceOf($token::class), $this->identicalTo(['ROLE_FOO']))
```

Comment:
Sorry if I'm picky here, but this assertion still does not check if the ADM is called with the right user. Instead it checks for the token class which is somewhat irrelevant for this functionality.

Reply:
Updated, I believe it should now better reflect what we actually want to verify.